modules/virtualisation/jails.nix

```diff
@@ -0,0 +1,362 @@
+              description = ''
+                Time for the container to start. In case of a timeout,
+                the container processes get killed.
+                See {manpage}`systemd.time(7)`
+                for more information about the format.
+               '';
+            };
+
+            ### Filesystem settings
+
+            bindMounts = mkOption {
+              type = with types; attrsOf (submodule bindMountOpts);
+              default = {
+                "/nix" = { isReadOnly = true; };
+                "/etc/resolv.conf" = { isReadOnly = true; isFile = true; };
+              };
+              example = literalExpression ''
+                { "/home" = { hostPath = "/home/alice";
+                              isReadOnly = false; };
+                }
+              '';
+
+              description = ''
+                  An extra list of directories that is bound to the container.
+                '';
+            };
+
+            tmpfs = mkOption {
+              type = types.listOf types.str;
+              default = [];
+              example = [ "/var" ];
+              description = ''
+                Mounts a set of tmpfs file systems into the container.
+                Multiple paths can be specified.
+                Valid items must conform to the --tmpfs argument
+                of systemd-nspawn. See {manpage}`systemd-nspawn(1)` for details.
+              '';
+            };
+
+            ### Network settings
+
+            vnet = mkOption {
+              type = types.bool;
+              default = false;
+              description = "Create a virtual network stack for the jail.";
+            };
+          };
+        }));
+
+      default = {};
+      example = literalExpression
+        ''
+          { webserver =
+              { path = "/nix/var/nix/profiles/webserver";
+              };
+            database =
+              { config =
+                  { config, pkgs, ... }:
+                  { services.postgresql.enable = true;
+                    services.postgresql.package = pkgs.postgresql_14;
+
+                    system.stateVersion = "${lib.trivial.release}";
+                  };
+              };
+          }
+        '';
+      description = ''
+        A set of NixBSD system configurations to be run as FreeBSD jails.
+      '';
+    };
+  };
+
+  config = mkMerge [
+    {
+      warnings = optional (!config.boot.enableJails && config.jails != {})
+        "jails.<name> is used, but boot.enableJails is false. To use jails.<name>, set boot.enableJails to true.";
+    }
+
+    (mkIf config.boot.enableJails {
+      freebsd.rc.services = lib.mapAttrs (name: cfg: let conf = mkJailConf name cfg; in {
+        name = "jail_${name}";
+        description = "Start the ${name} jail specified in the system configuration.";
+        path = [ pkgs.freebsd.jail ] ++ config.environment.systemPackages;
+        rcorderSettings.REQUIRE = [ "mini_tmpfiles" "NETWORKING" ];
```

Comment:
I'd rather keep the name "mini_tmpfiles" as an implementation detail. We could have a "tempfiles" provide in mini_tmpfiles